packages/jest-fake-timers/src/__tests__/jestFakeTimers.test.ts

```diff
@@ -651,6 +651,100 @@ describe('FakeTimers', () => {
     });
   });
 
+  describe('advanceTimersToNextTimer', () => {
```

Comment:
can you add a test where a function adds another timer?

```js
const runOrder: Array<string> = [];
const mock1 = jest.fn(() => runOrder.push('mock1'));
const mock2 = jest.fn(() => runOrder.push('mock2'));
const mock3 = jest.fn(() => runOrder.push('mock3'));
const mock4 = jest.fn(() => runOrder.push('mock4'));

global.setTimeout(mock1, 0);
global.setTimeout(() => {
  mock2();
  global.setTimeout(mock3, 50)
}, 25);
global.setTimeout(mock4, 100);

timers.advanceTimersToNextTimer(3);
```

I'm not sure if I expect `runOrder` to be `['mock1', 'mock2', 'mock3']` or `['mock1', 'mock2', 'mock3', 'mock4']`. What do you think? I'm _leaning_ towards all 4 (so you say three steps, we decide how far to advance the timer, and fire all timers added as we go within that timeframe, including newly added ones even though that increases the number of steps).

(Note that my suggested implementation of a single `sort` then `advance` would result in all 4, but maybe we want just `1, 2, 3`?)

Reply:
`advanceTimersToNextTimer` should let you assert in your test between `mock2` and `mock3`, that's way the run order will be `['mock1', 'mock2', 'mock3']`

Reply:
Agreed!